pkg/envoy/lds/inmesh.go

```diff
@@ -116,6 +116,15 @@ func (lb *listenerBuilder) getInboundMeshHTTPFilterChain(proxyService service.Me
 	}
 
 	filterchainName := fmt.Sprintf("%s:%s:%d", inboundMeshHTTPFilterChainPrefix, proxyService, servicePort)
+
+	serverNames := []string{proxyService.ServerName()}
+	// The sidecar needs to be aware of the names a remote cluster might refer to this service as.
+	if lb.cfg.GetFeatureFlags().EnableMulticlusterMode {
```

Comment:
Could you add a `proxyKind` field to the `listenerBuilder` struct and add an additional check of `if lb.proxyKind == envoy.KindGateway` in L122.

We don't want the additional server name processing for in-mesh clients that are not a gateway.

Reply:
the gateway does TLS passthrough to the upstream service. The downstream service refers to the upstream via:

service.namespace.svc.cluster.cluster-id, so the upstream service will need the additional server name to process. The only other alternative is to not do SSL Passthrough on the gateway, which would be a large rewrite

Reply:
Okay sounds good.